application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,165 @@
+
+        getData().addOptions(messageOption, guildOption, anonymousOption);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+        event.deferReply().setEphemeral(true).queue();
+
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        authorIdToLastCommandInvocation.put(userId, Instant.now());
+
+        Optional<TextChannel> modMailAuditLog = getChannel(event);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn("Cannot find the designated modmail channel in this server.");
+            return;
+        }
+
+        String userMessage = event.getOption(OPTION_MESSAGE).getAsString();
+        boolean wantsToStayAnonymous = event.getOption(OPTION_STAY_ANONYMOUS).getAsBoolean();
+        String user = event.getUser().getAsMention();
+        MessageCreateAction message = modMailAuditLog.orElseThrow()
+            .sendMessageEmbeds(createModMailMessage(user, userMessage))
+            .addActionRow(DiscordClientAction.General.USER.asLinkButton("Click to see profile!",
+                    String.valueOf(userId)));
+
+        if (wantsToStayAnonymous) {
+            user = "Anonymous";
+            message = modMailAuditLog.orElseThrow()
+                .sendMessageEmbeds(createModMailMessage(user, userMessage));
+        }
+
+        message.mapToResult().flatMap(result -> {
+            if (result.isSuccess()) {
+                return event.getHook().editOriginal("Your message has been forwarded, thanks.");
+            } else {
+                return event.getHook()
+                    .editOriginal("There was an issue forwarding your message, sorry. We are "
+                            + "investigating.");
+            }
+        }).queue();
+
+    }
+
+    private MessageEmbed createModMailMessage(String user, String userMessage) {
+        return new EmbedBuilder().setDescription("**/modmail from %s** ".formatted(user))
+            .setFooter(userMessage)
+            .setColor(AMBIENT_COLOR)
+            .build();
+    }
+
+    private Optional<TextChannel> getChannel(SlashCommandInteractionEvent event) {
+        long userGuildId = event.getOption(OPTION_SERVER_GUILD).getAsLong();
+        Guild guild = Objects.requireNonNull(event.getJDA().getGuildById(userGuildId),
+                "Something went wrong with selecting the guild.");
```

Comment:
what went wrong? that error message isnt helpful. it is actually not even necessary, u can just drop the whole requireNonNull check since you use the variable in the next line anyways.